roles/telemetry_autoscaling/templates/template.yaml.j2

```diff
@@ -0,0 +1,120 @@
+heat_template_version: wallaby
+description:  Example auto scale group, policy and alarm
+parameters:
+  server_name_prefix:
+    description: A prefix for servers created by this stack. Can be used in queries.
+    type: string
+    default: autoscaling_server_
+
+resources:
+  scaleup_group:
+    type: OS::Heat::AutoScalingGroup
+    properties:
+      max_size: 3
+      min_size: 1
+      desired_capacity: 1
+      resource:
+        type: OS::Nova::Server::VNF
+        properties:
+          server_name_prefix: { get_param: server_name_prefix }
+          metadata: {"metering.server_group": {get_param: "OS::stack_id"}}
+
+
+  scaleup_policy:
+    type: OS::Heat::ScalingPolicy
+    properties:
+      adjustment_type: change_in_capacity
+      auto_scaling_group_id: { get_resource: scaleup_group }
+      cooldown: 60
+      scaling_adjustment: 1
+
+  scaledown_policy:
+    type: OS::Heat::ScalingPolicy
+    properties:
+      adjustment_type: change_in_capacity
+      auto_scaling_group_id: { get_resource: scaleup_group }
+      cooldown: 60
+      scaling_adjustment: -1
+
+  cpu_alarm_high:
+    {% if metrics_backend == "gnocchi" -%}
+    type: OS::Aodh::GnocchiAggregationByResourcesAlarm
+    {% endif -%}
+    {% if metrics_backend == "prometheus" -%}
+    type: OS::Aodh::PrometheusAlarm
+    {% endif -%}
+    properties:
+      description: Scale up instance if CPU > 50%
+      {% if metrics_backend == "gnocchi" -%}
+      metric: cpu
+      aggregation_method: rate:mean
+      granularity: 300
+      evaluation_periods: 1
+      resource_type: instance
+      threshold: 30000000000.0
+      {% endif -%}
+      {% if metrics_backend == "prometheus" -%}
+      threshold: 50
+      {% endif -%}
+      comparison_operator: gt
+      alarm_actions:
+        - str_replace:
+            template: trust+url
+            params:
+              url: {get_attr: [scaleup_policy, signal_url]}
+      query:
+        {% if metrics_backend == "gnocchi" -%}
+        list_join:
+          - ''
+          - - {'=': {server_group: {get_param: "OS::stack_id"}}}
+        {% endif -%}
+        {% if metrics_backend == "prometheus" -%}
+        str_replace:
+          template: "(rate(ceilometer_cpu{resource_name=~'server_name_prefix.*'}[150s]))/10000000"
```

Comment:
maybe its better if we updated the query with the latest change we have in tempest, https://review.opendev.org/c/openstack/telemetry-tempest-plugin/+/920015
there was some failure cases where the rate function returned 0

Reply:
This follows the documented procedure that Jaromir and Michael shared. 

The queries in TTP are of a different magnitude, with a configurable threshold, and a differing period:

"query": "(rate(ceilometer_cpu{resource_name=~'te-$ENVIRON['RESOURCE_PREFIX'].*'}[$ENVIRON['PROMETHEUS_RATE_DURATION']s])) * 100"

Reply:
yes we are using the rate function, but his update cover the duration used by the prometheus.
for stable testing it should be <ceilometer_polling_interval> +<prometheus_scrape_interval>. With other static values  it's possible,that the rate() function will just return 0. and that what happened in downstream with tempest test.

Reply:
my suggestion is to have something similar, for example  to have the values as variables with default parameter  we can change if we faced a problem.   

prometheus_rate_duration = (config.CONF.telemetry.ceilometer_polling_interval + config.CONF.telemetry.prometheus_scrape_interval)

    cfg.IntOpt('ceilometer_polling_interval',
               default=300,
               help="Polling interval configured for ceilometer. This can "
                    "be used in test cases to wait for metrics to appear."),
    cfg.IntOpt('prometheus_scrape_interval',
               default=15,
               help="Scrape interval configured for prometheus. This can "
                    "be used in test cases to properly configure autoscaling")

Reply:
I see. I think this warrants some additional attention and is outside the scope of this change. I'll file a bug for follow-up, so we can also make sure that the documentation covers this information.